routemaster/state_machine/types.py

```diff
@@ -1,8 +1,16 @@
 """Shared types for state machine execution."""
 
-from typing import Any, Dict, NamedTuple
+from typing import Any, Dict, Callable, NamedTuple
+
+from routemaster.app import App
+from routemaster.config import State, StateMachine
 
 Metadata = Dict[str, Any]
+IsExitingCheck = Callable[[], bool]
+CronProcessor = Callable[
```

Comment:
Shame we don't have a top-level types file for these.

Reply:
I'm not sure it should go in there. I might drop the cron specificity here though.

Reply:
2504fbcae2cc1befeecc6f8d8918e86a3297e16c